implants/lib/eldritch/src/assets/read_binary_impl.rs

```diff
@@ -1,6 +1,31 @@
-use anyhow::Result;
+use std::sync::mpsc::{channel, Receiver};
 
-pub fn read_binary(src: String) -> Result<Vec<u32>> {
+use anyhow::{Context, Result};
+use pb::c2::FetchAssetResponse;
+use starlark::{eval::Evaluator, values::list::ListRef};
+
+use crate::runtime::{messages::FetchAssetMessage, Environment};
+
+fn read_binary_remote(rx: Receiver<FetchAssetResponse>) -> Result<Vec<u32>> {
+    // Wait for our first chunk
+    let resp = rx.recv()?;
```

Comment:
is there a reason we do the first chunk separately?

Reply:
I don't think so tests pass without it.